android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandlerOrchestrator.java

```diff
@@ -485,9 +486,16 @@ private static boolean canRunSimultaneously(GestureHandler a, GestureHandler b)
   }
 
   private static boolean shouldHandlerBeCancelledBy(GestureHandler handler, GestureHandler other) {
+
     if (canRunSimultaneously(handler, other)) {
       return false;
-    } else if (handler != other &&
+    }
+    if (handler.getState() == GestureHandler.STATE_BEGAN && !handler.hasSameFirstPointerId(other)) {
```

Comment:
- [ ] remove unnecessary check for BEGAN
 - [ ] test if handler and other use no common pointers instead of only checking first pointer id